resources/ext/cli_defaults/cli-defaults.sh.erb

```diff
@@ -4,4 +4,20 @@ if [ -n "$JRUBY_JAR" ]; then
   echo "Warning: the JRUBY_JAR setting is no longer needed and will be ignored." 1>&2
 fi
 
+java_version=$($JAVA_BIN -version 2>&1 | head -1 | awk -F\" '{ print $2 }')
+java_major_version=$(echo $java_version | awk -F. '{ print $1 }')
+
+if [[ $java_major_version -eq 17 ]]; then
+
+	echo $JAVA_ARGS | grep "add-opens" &>/dev/null
```

Comment:
The title might be a bit of a misnomer as we're only opening the IO subsystem if the user isn't already messing with opening subsystems. It feels like at that point the user either has this covered or can update it themselves.